api/v1beta1/vmcluster_types.go

```diff
@@ -948,17 +952,16 @@ func (cr *VMCluster) LastAppliedSpecAsPatch() (client.Patch, error) {
 	return client.RawPatch(types.MergePatchType, []byte(patch)), nil
 }
 
-// GetLastAppliedSpec returns last applied cluster spec
-func (cr *VMCluster) GetLastAppliedSpec() (*VMClusterSpec, error) {
+// CompareSpecChange compares cluster spec with last applied cluster spec stored in annotation
+func (cr *VMCluster) CompareSpecChange() ([]string, error) {
 	var prevClusterSpec VMClusterSpec
-	prevClusterJSON := cr.Annotations["operator.victoriametrics/last-applied-spec"]
-	if prevClusterJSON == "" {
-		return &prevClusterSpec, nil
+	lastAppliedClusterJSON := cr.Annotations["operator.victoriametrics/last-applied-spec"]
+	if err := json.Unmarshal([]byte(lastAppliedClusterJSON), &prevClusterSpec); err != nil {
+		return []string{""}, fmt.Errorf("cannot parse last applied cluster spec value: %s : %w", lastAppliedClusterJSON, err)
 	}
-	if err := json.Unmarshal([]byte(prevClusterJSON), &prevClusterSpec); err != nil {
-		return nil, fmt.Errorf("cannot parse last applied cluster spec value: %s : %w", prevClusterJSON, err)
-	}
-	return &prevClusterSpec, nil
+	instanceSpecData, _ := json.Marshal(cr.Spec)
+	changes := deep.Equal([]byte(lastAppliedClusterJSON), instanceSpecData)
```

Comment:
Probably, we could use just simple `bytes.Compare` or check string equality, since `deep.Equal` checks structs.